papas/simulation/src/StraightLinePropagator.cpp

```diff
@@ -1,27 +1,26 @@
 
 
 //#include <iostream>
-#include "papas/simulation/StraightLinePropagator.h"
 #include "papas/datatypes/Helix.h"
 #include "papas/datatypes/PFParticle.h"
 #include "papas/datatypes/Path.h"
+#include "papas/simulation/StraightLinePropagator.h"
 #include "papas/utility/GeoTools.h"
 
 namespace papas {
 
 StraightLinePropagator::StraightLinePropagator() {}
 
-void StraightLinePropagator::propagateOne(const PFParticle& ptc,
-                                          papas::Position layer,
-                                          double cylinderz,
-                                          double cylinderradius) const {
+void StraightLinePropagator::propagateOne(const PFParticle& ptc, const SurfaceCylinder& cyl, double field) const {
+  auto layer = cyl.layer();
+  auto cylinderz = cyl.z();
+  double cylinderradius = cyl.radius();
```

Comment:
why auto in one case and double in the other? 
for native types I prefer explicit typing